src/VisualStudio/Core/Next/Remote/ServiceHubRemoteHostClient.cs

```diff
@@ -74,17 +74,23 @@ internal sealed partial class ServiceHubRemoteHostClient : RemoteHostClient
                 return;
             }
 
-            // RegisterWorkspaceHost is required to be called from UI thread so push the code
-            // to UI thread to run. 
-            await Task.Factory.SafeStartNew(() =>
+            // Create a connection to the host in the BG to avoid taking the hit of loading 
+            // service hub on the UI thread.
+            var currentConnection = await client.TryCreateConnectionAsync(WellKnownRemoteHostServices.RemoteHostService, CancellationToken.None).ConfigureAwait(false);
```

Comment:
💭 It would be nice for `TryCreateConnectionAsync` to return an RCD, but that's for a later change.